### ISSUE TITLE
[C+ Checklist Needs Completion] [$250] Introduce a new hasPendingOnyxUpdates flag to push notifications payload

### ISSUE THREAD

**Issue by rlinoz** [Expensify/App#53643]
**Problem**: When the push notification payload is too big we don't send onyx updates, and when we don't send onyx updates the app doesn't fetch new data in the background when receiving a new notification, [see src code](https://github.com/Expensify/App/blob/55a1f7d57b0b06724295be3c6136360d803b834b/src/libs/Notification/PushNotification/subscribePushNotification/index.ts#L50).

**Solution**: Update this code https://github.com/Expensify/Web-Expensify/blob/ffc9c9b04976154da86d491209421a326aa961db/lib/MobilePushNotifications.php#L83 to send a new `hasPendingOnyxUpdates` flag when we remove the onyx updates from the notification and update the FE code to use that flag to fetch the newest updates upon receiving the notification.

**More context**: https://expensify.slack.com/archives/C03TQ48KC/p1733411952819489?thread_ts=1733409200.404039&cid=C03TQ48KC
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021876999249012730708</li>
        <li>Upwork Job ID: 1876999249012730708</li>
        <li>Last Price Increase: 2025-02-01</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @greg-schroeder </details>

**Comment by rlinoz** [comment]
Backend PR is up, asking for some help with the frontend here https://expensify.slack.com/archives/C035J5C9FAP/p1734100891197679

**Comment by rlinoz** [comment]
Will open up for proposals if no expert picks it up

**Comment by chrispader** [comment]
Hey, i'm Chris from Margelo, a RN expert agency. commenting for assignment! :)

**Comment by chrispader** [comment]
@rlinoz could you provide me access or a brief context about what has been discussed here? [expensify.slack.com/archives/C03TQ48KC/p1733411952819489?thread_ts=1733409200.404039&cid=C03TQ48KC](https://expensify.slack.com/archives/C03TQ48KC/p1733411952819489?thread_ts=1733409200.404039&cid=C03TQ48KC)

**Comment by rlinoz** [comment]
Sure!

For notifications where we drop the Onyx payload because they are too big we now include a flag `hasPendingOnyxUpdates` and also the `lastUpdateID` available.

The idea is if the flag is true and we have a `lastUpdateID` call `GetMissingOnyxMessages` using `updateIDFrom=<last client update id>` and `updateIDTo=<the lastUpdateID we just received in the push>`.

@chrispader does that make sense?

**Comment by chrispader** [comment]
> Sure!
> 
> For notifications where we drop the Onyx payload because they are too big we now include a flag `hasPendingOnyxUpdates` and also the `lastUpdateID` available.
> 
> The idea is if the flag is true and we have a `lastUpdateID` call `GetMissingOnyxMessages` using `updateIDFrom=<last client update id>` and `updateIDTo=<the lastUpdateID we just received in the push>`.
> 
> @chrispader does that make sense?

Yes, thanks that's exactly what i needed. With the `lastUpdateID` we can easily trigger the fetching for the missing updates.

**Comment by chrispader** [comment]
@rlinoz when the `hasPendingOnyxUpdates` flag is set to true, might there still be some/other Onyx updates in the payload, or will the `onyxData` always be undefined?

**Comment by rlinoz** [comment]
@chrispader `onyxData` and `previousUpdateID` will always be undefined if `hasPendingOnyxUpdates` is true

btw, the change is not live yet, although it is merged so should be deployed to staging soon

**Comment by chrispader** [comment]
> @chrispader `onyxData` and `previousUpdateID` will always be undefined if `hasPendingOnyxUpdates` is true

thanks!

> btw, the change is not live yet, although it is merged so should be deployed to staging soon

yes, i'm aware of that :)

**Comment by chrispader** [comment]
@rlinoz just created a PR here: https://github.com/Expensify/App/pull/54257

Will finalize later today or tmrw morning

**Comment by greg-schroeder** [comment]
yoooo 👋  

Oh wow, what a gift, an issue that's already underway with merged PRs 😎

**Comment by greg-schroeder** [comment]
@eVoloshchak nearing the payment date here - can you finish up the checklist?

**Comment by greg-schroeder** [comment]
Payment summary:

Contributor: @chrispader - the hobby horse legend himself! - no payment required for this one
C+: @eVoloshchak - $250 - You can make a manual request via ND, though you still have to complete the [checklist](https://github.com/Expensify/App/issues/53643#issuecomment-2585044341) <3

**Comment by greg-schroeder** [comment]
bump @eVoloshchak

**Comment by rlinoz** [comment]
@eVoloshchak can you complete the checklist please?

**Comment by JmillsExpensify** [comment]
$250 approved for @eVoloshchak

**Comment by greg-schroeder** [comment]
Thanks @eVoloshchak
